UPGRADE.md

```diff
@@ -24,6 +24,30 @@ Removed unused arguments:
 
 - `Sulu\Component\Webspace\Analyzer\Attributes\WebsiteRequestProcessor::__construct` `$contentMapper` (2nd argument) removed
 
+### Updating to flysystem 3
+The Sulu image storage implementation is now only supported by configuring the Flysystem adapter like this:
+```yaml
+# config/packages/flysystem.yaml
+flysystem:
+    storages:
+        default.storage:
+            adapter: 'local'
+            options:
+                directory: '%kernel.project_dir%/var/uploads'
+```
+The service defined there has to referenced in the `SuluMediaBundle` config:
+```yaml
+# config/packages/sulu_media.yaml
+sulu_media:
+    storage:
+        service: 'default.storage'
+        segment: 10
+```
+This will only create the service `sulu_media.storage` as the alias to `sulu_media.storage.default` has been removed.
+
+Other removed services:
+- `Sulu\\Bundle\\MediaBundle\\DependencyInjection\\S3ClientCompilerPass` (internal)
```

Comment:
````suggestion
### Updating to flysystem 3

The Sulu image storage implementation uses now Flysystem over the Flysystem Bundle:

```yaml
# config/packages/flysystem.yaml
flysystem:
    storages:
        default.storage:
            adapter: 'local'
            options:
                directory: '%kernel.project_dir%/var/uploads'
```

If you want use a different storage for Sulu you can configure it in:

```yaml
# config/packages/sulu_media.yaml
sulu_media:
    storage:
        service: 'default.storage' # this is default and not required to be configured.
```

This will only create the service `sulu_media.storage` as the alias to `sulu_media.storage.*` services has been removed.
````

Add `S3ClientCompilerPass`, `LocalStorage`, `S3Storage`, ... just to the removed classes, services list which already exists.